src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidCharacterNameTagCheck.java

```diff
@@ -0,0 +1,109 @@
+    private final Set<String> localizedNameTags = new HashSet<>();
+    private final TaggableFilter validObjectFilter;
+
+    /**
+     * Default constructor
+     *
+     * @param configuration
+     *            {@link Configuration} required to construct any Check
+     */
+    public InvalidCharacterNameTagCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.validObjectFilter = configurationValue(configuration, "valid.object.filter",
+                VALID_OBJECT_FILTER_DEFAULT, TaggableFilter::forDefinition);
+        final Set<Optional<IsoLanguage>> languages = IsoLanguage.allLanguageCodes().stream()
+                .map(languageCode -> Optional.of(IsoLanguage.forLanguageCode(languageCode).get()))
+                .collect(Collectors.toSet());
+        // The empty optional is added here to include the non localized versions, ie. "name",
+        // along with the localized versions.
+        languages.add(Optional.empty());
+        for (final Optional<IsoLanguage> language : languages)
+        {
+            Validators.localizeKeyName(NameTag.class, language)
+                    .ifPresent(this.localizedNameTags::add);
+        }
+    }
+
+    /**
+     * Checks to see whether the supplied object class type is valid for this particular check
+     *
+     * @param object
+     *            The {@link AtlasObject} you are checking
+     * @return true if it is
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return (object instanceof Area || object instanceof Line || object instanceof Relation)
+                && this.validObjectFilter.test(object);
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Set<String> invalidCharacterNameTags = new HashSet<>();
+        final Optional<String> nameTag = NameTag.getNameOf(object);
+        // If NameTag is present and its value matches the invalid character pattern, add the Name
+        // tag to the invalidCharacterNameTags set.
+        if (nameTag.isPresent() && INVALID_CHARS_REGEX_DEFAULT.matcher(nameTag.get()).matches())
+        {
+            invalidCharacterNameTags.add(NameTag.KEY);
+
+        }
+        // If the atlas object has localized name tags that matches the invalid character pattern,
+        // add it to the invalidCharacterNameTags set.
+        Iterables.stream(object.getOsmTags().keySet()).filter(this.localizedNameTags::contains)
+                .filter(localizedNameTag -> INVALID_CHARS_REGEX_DEFAULT
+                        .matcher(object.getTag(localizedNameTag).get()).matches())
+                .forEach(invalidCharacterNameTags::add);
+
+        return invalidCharacterNameTags.isEmpty() ? Optional.empty()
+                : Optional.of(this.createFlag(object, this.getLocalizedInstruction(0,
```

Comment:
It would be good to flag Edges with the same OSM ID together.

Reply:
Current code does not include Edges, only considers Areas, Lines and Relation.

Reply:
Oh yep, my bad.